pkg/azurekeyvault/client/certificate.go

```diff
@@ -183,7 +184,18 @@ func importPem(pemCert string) (*Certificate, error) {
 		raw = rest
 	}
 
-	cert.Certificates, err = x509.ParseCertificates(publicDers)
+	if ensureServerFirst && len(publicDers) > 1 {
+		publicDers = append(
+			publicDers[len(publicDers)-1:],
+			publicDers[0:len(publicDers)-1]...,
+		)
+	}
+
+	for _, der := range publicDers {
+		joinedPublicDers = append(joinedPublicDers, der...)
+	}
+
+	cert.Certificates, err = x509.ParseCertificates(joinedPublicDers)
```

Comment:
I realised the reordering needed to be done in importPem and not NewCertificateFromPfx since the key can be last entry. This would probably not be an issue since the Key Vault export seems to like to put the key first but there could be scenarios where that is not the case.